core/src/main/scala-3/pureconfig/generic/derivation/EnumConfigConvertDerivation.scala

```diff
@@ -0,0 +1,48 @@
+package pureconfig
+package generic
+package derivation
+
+import scala.compiletime.{constValue, erasedValue, error, summonInline}
+import scala.deriving.Mirror
+
+import com.typesafe.config.ConfigValue
+
+import pureconfig.error.{CannotConvert, ConfigReaderFailures}
+import pureconfig.generic.derivation.Utils._
+
+type EnumConfigConvert[A] = EnumConfigConvertDerivation.Default.EnumConfigConvert[A]
+
+trait EnumConfigConvertDerivation extends EnumConfigReaderDerivation, EnumConfigWriterDerivation {
+  trait EnumConfigConvert[A] extends ConfigConvert[A]
+  object EnumConfigConvert {
+    inline def derived[A: Mirror.SumOf]: EnumConfigConvert[A] =
+      new EnumConfigConvert[A] {
+        val reader = EnumConfigReader.derived[A]
+        val writer = EnumConfigWriter.derived[A]
+
+        def from(cur: ConfigCursor): ConfigReader.Result[A] = reader.from(cur)
+        def to(a: A): ConfigValue = writer.to(a)
+      }
+  }
+}
+
+object EnumConfigConvertDerivation {
+  private lazy val defaultMapping = ConfigFieldMapping(PascalCase, KebabCase)
+
+  object Default
+      extends EnumConfigConvertDerivation,
+        EnumConfigReaderDerivation(defaultMapping),
+        EnumConfigWriterDerivation(defaultMapping)
+
+  inline def deriveEnumerationConvert[A: Mirror.SumOf](transformName: String => String): ConfigConvert[A] = {
+    object Instance
+        extends EnumConfigConvertDerivation,
+          EnumConfigReaderDerivation(transformName),
+          EnumConfigWriterDerivation(transformName)
+
+    Instance.EnumConfigConvert.derived[A]
+  }
+
+  inline def deriveEnumerationConvert[A: Mirror.SumOf]: ConfigConvert[A] =
+    Default.EnumConfigConvert.derived[A]
```

Comment:
I would prefer if we would keep these `derive*` methods exposed only in pureconfig-generic-scala3. We're going to include both pureconfig-core and pureconfig-generic-scala3 in the pureconfig bundle, so it can be confusing for downstream users of the bundle to have the multiple locations.

Reply:
Makes sense. I pushed all of them to `EnumDerivationUtils` object in `pureconfig-generic-scala3`.